mock-connector/src/main/scala/it/agilelab/darwin/connector/mock/MockConnector.scala

```diff
@@ -1,32 +1,92 @@
 package it.agilelab.darwin.connector.mock
 
 import com.typesafe.config.Config
-import it.agilelab.darwin.common.Connector
-import org.apache.avro.Schema
+import it.agilelab.darwin.common.compat._
+import it.agilelab.darwin.common.{Connector, Logging, SchemaReader}
+import org.apache.avro.{Schema, SchemaNormalization}
 
 import scala.collection.mutable
 
-class MockConnector(config: Config) extends Connector {
+class MockConnectorException(msg: String, t: Option[Throwable]) extends RuntimeException(msg) {
+  def this(msg: String) = this(msg, None)
 
-  val table: mutable.Map[Long, Schema] = {
-    val alone = parseResource("test/MockClassAlone.avsc")
-    val parent = parseResource("test/MockClassParent.avsc")
-    mutable.Map(0L -> alone, 1L -> parent)
+  def this(t: Throwable) = this(t.getMessage, Some(t))
+
+  override def getCause: Throwable = t match {
+    case Some(value) => value
+    case None => super.getCause
+  }
+}
+
+class MockConnector(config: Config) extends Connector with Logging {
+
+  val mode: ConfigurationKeys.Mode = if (config.hasPath(ConfigurationKeys.MODE)) {
+    ConfigurationKeys.Mode.parse(config.getString(ConfigurationKeys.MODE))
+  } else {
+    ConfigurationKeys.Strict
+  }
+
+  private val files = if (config.hasPath(ConfigurationKeys.FILES)) {
```

Comment:
I would make files, resources and table lazy variables so the loading errors occur on actual data access and not on instance construnction.

Reply:
I don't like to make them lazy, because that adds a semaphore to the variables. I think I'm making them methods and calling them in `fullLoad()`